odoo/addons/base/models/ir_async.py

```diff
@@ -0,0 +1,265 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import datetime
+import itertools
+import json
+import logging
+import re
+import traceback
+import threading
+import odoo
+import odoo.exceptions
+from odoo import models, fields, api
+from odoo.http import serialize_exception
+
+_logger = logging.getLogger(__name__)
+BASE_VERSION = odoo.modules.load_information_from_description_file('base')['version']
+
+
+class IrAsync(models.Model):
+    _name = 'ir.async'
+    _description = 'Asynchronous Jobs'
+
+    name = fields.Char()
+    state = fields.Selection([
+        ('created', 'Created'),     # The job has been enqueued for later processing
+        ('succeeded', 'Succeeded'), # The job finished with a result, the result has not yet been retrieved
+        ('failed', 'Failed'),       # The job failed due to an exception
+        ('done', 'Done'),           # The job finished without result or the result has been retrieved
+    ])
+    res_model = fields.Char()
+    res_ids = fields.Text()
+    method = fields.Char()
+    args = fields.Text()
+    kwargs = fields.Text()
+    user_id = fields.Many2one('res.users', string='Scheduler User', ondelete='cascade')
```

Comment:
Shouldn't it be `required` ? Can we have `ir_async` records without a `user_id` ?

Reply:
All the fields that are actually required for proper operation need to be marked as `required`. It's not useless, it's necessary for the integrity of the schema.

Reply:
Is a `user_id` field even useful? Wouldn't the `create_uid` be the same value now that sudo mode keeps the current user around?

Reply:
Also @odony maybe only the state field and possibly the description should be user-readable?